rust_snuba/src/strategies/python_v2.rs

```diff
@@ -0,0 +1,290 @@
+        for py_message in messages {
+            let (
+                (payload, origin_timestamp, sentry_received_timestamp),
+                message_timestamp,
+                offsets,
+            ) = py_message;
+
+            let commit_log_offsets = offsets
+                .iter()
+                .map(|((_t, p), o)| (*p, (*o, message_timestamp)))
+                .collect();
+
+            let payload = BytesInsertBatch::new(
+                RowData::from_encoded_rows(payload),
+                message_timestamp,
+                origin_timestamp,
+                sentry_received_timestamp,
+                commit_log_offsets,
+            );
+
+            let mut committable: BTreeMap<Partition, u64> = BTreeMap::new();
+            for ((t, p), o) in offsets {
+                committable.insert(Partition::new(Topic::new(&t), p), o);
+            }
+
+            let message = Message::new_any_message(payload, committable);
+            self.transformed_messages.push_back(message);
+        }
+    }
+}
+
+impl ProcessingStrategy<KafkaPayload> for PythonTransformStep {
+    fn poll(&mut self) -> Result<Option<CommitRequest>, InvalidMessage> {
+        let messages = Python::with_gil(|py| -> PyResult<Vec<PyReturnValue>> {
+            let python_strategy = self.python_strategy.lock();
+            let result = python_strategy.call_method0(py, "poll")?;
+            Ok(result.extract(py).unwrap())
+        })
+        .unwrap();
+
+        self.handle_py_return_value(messages);
+
+        // Attempt to submit all transformed messages
+        while let Some(msg) = self.transformed_messages.pop_front() {
+            let commit_request = self.next_step.poll()?;
+            self.commit_request_carried_over =
+                merge_commit_request(self.commit_request_carried_over.take(), commit_request);
+
+            match self.next_step.submit(msg) {
+                Err(SubmitError::MessageRejected(MessageRejected {
+                    message: transformed_message,
+                })) => {
+                    self.transformed_messages.push_front(transformed_message);
+                    break;
+                }
+                Err(SubmitError::InvalidMessage(invalid_message)) => {
+                    return Err(invalid_message);
+                }
+                Ok(_) => {}
+            }
+        }
+
+        let commit_request = self.next_step.poll()?;
+
+        Ok(merge_commit_request(
+            self.commit_request_carried_over.take(),
+            commit_request,
+        ))
+    }
+
+    fn submit(&mut self, message: Message<KafkaPayload>) -> Result<(), SubmitError<KafkaPayload>> {
+        if self.transformed_messages.len() > 100_000 {
+            return Err(SubmitError::MessageRejected(MessageRejected { message }));
+        }
+
+        let mut apply_backpressure = false;
+
+        let _ = Python::with_gil(|py| -> PyResult<()> {
```

Comment:
please handle this result. you can probably just unwrap here

Reply:
Updated